### ISSUE TITLE
DTI fitting function with NLLS method raises an error.

### ISSUE THREAD

**Issue by GalKepler** [dipy/dipy#2790]
## Description
Changes introduced to the DTI fitting function for NLLS method raise an error.

I've narrowed the issue to a single line of code that was changed sometime in the last few months:
In [here](https://github.com/dipy/dipy/blob/master/dipy/reconst/dti.py#L1724), the OLS parameters were previously estimated as:

```python
inv_design = np.linalg.pinv(design_matrix)
log_s = np.log(flat_data)
D = np.dot(inv_design, log_s.T).T
```
This behavior changed into:
````python
D = ols_fit_tensor(design_matrix, flat_data, return_lower_triangular=True)
````

Which, for some reason, raises the following error:
```python
ValueError: could not broadcast input array from shape (10000,7) into shape (10000,12)
```
Changing this line of code to what it previously was fixed this issue (while passing the same inputs for DTI reconstruction), but I'm not sure why was this change introduced in the first place so I'm not sure which is the correct way to go.

## Way to reproduce
- [X] Code example
```python
from dipy.workflows.reconst import ReconstDtiFlow
a = {"input_files" : path_to_dwi_nifti,
"bvalues_files": path_to_dwi_bval,
"bvectors_files": path_to_dwi_bvec,
"mask_files": path_to_dwi_mask,
"out_dir": path_to_output_directory,
"fit_method": "NLLS"}

runner = ReconstDtiFlow()
runner.run(**a)
```

- [X] Operating system and version (run `python -c "import platform; print(platform.platform())"`)
Linux-5.10.0-21-amd64-x86_64-with-glibc2.31
- [X] Python version (run `python -c "import sys; print("Python", sys.version)"`)
3.10.2
- [X] dipy version (run `python -c "import dipy; print(dipy.__version__)"`)
1.7.0

**Comment by skoudoro** [comment]
Hi @GalKepler,

Thank you for your feedback.

@samcoveney, can you explain your recent change to @GalKepler. He had some issue. 

Thank you

**Comment by samcoveney** [comment]
sorry @GalKepler and @skoudoro  I completely missed this conversation.

I haven't seen this error myself, and I use the NLLS fitting all the time.

I am having some problems reproducing this at the moment, since I keep seeing the error

```
ModuleNotFoundError: No module named 'dipy.direction.ptt_direction_getter'
```

when trying to run this workflow, or when trying to run some of the tests. But the NLLS fitting routines are working for me.

Is it only when using this workflow that you see this issue?

**Comment by samcoveney** [comment]
Alright, it seems to be an issue with ```iter_fit_tensor``` decorator, which expects the wrapped function to return an array of size 12.

But when using ```return_lower_triangular=True``` the wrapped function returns an array of size 7 instead...

Two approaches would fix this:
(1) allow ```iter_fit_tensor``` to handle returns smaller than size 12
(2) undo some of the changes

Sorry about this, all the tests passed, I guess this was just missed. The tests are small, and so did not trigger use of the ```iter_fit_tensor``` wrapper. And my own work never highlighted this problem either.

I can try to implement a fix and put in a pull request today

**Comment by samcoveney** [comment]
This should fix everything, sorry again about the mistake

**Comment by skoudoro** [comment]
closed by #2730. Thanks again @samcoveney
